apps/system/js/tracking_notice.js

```diff
@@ -0,0 +1,152 @@
+/* global SystemDialog, LazyLoader, Service */
+'use strict';
+
+(function(exports) {
+
+  const TRACKING_NOTICE_KEY = 'privacy.trackingprotection.shown';
+  const TRACKING_NOTICE_LEARN_MORE_URL = 'tracking-notice-learn-more-url';
+
+  /**
+   * @class TrackingNotice
+   * @param {options} object for attributes `onShow`, `onHide` callback.
+   * @extends SystemDialog
+   */
+  var TrackingNotice = function(controller) {
+    this.instanceID = 'tracking-notice';
+    this.controller = controller || {};
+    this.options = {};
+    this._settings = navigator.mozSettings;
+    var req = navigator.mozSettings.createLock().get(TRACKING_NOTICE_KEY);
+    req.onsuccess = (function() {
+      var alreadyShown = req.result[TRACKING_NOTICE_KEY];
+      this._alreadyShown = alreadyShown;
+      if (!alreadyShown) {
+        this.renderDialog();
+      }
+    }.bind(this));
+
+    req.onerror = (function() {
+      this._alreadyShown = false;
+      this.renderDialog();
+    }.bind(this));
+
+    this.publish('created');
+  };
+
+  TrackingNotice.prototype = Object.create(SystemDialog.prototype);
+
+  TrackingNotice.prototype.customID = 'tracking-notice';
+
+  TrackingNotice.prototype.DEBUG = false;
+
+  TrackingNotice.prototype.name = 'TrackingNotice';
+
+  TrackingNotice.prototype.containerElement =
+    document.getElementById('fullscreen-overlay');
+
+  TrackingNotice.prototype.renderDialog = function() {
+    LazyLoader.load('shared/elements/gaia_switch/script.js', function() {
+      this.render();
+      this.boundShowIfNeeded = this.showIfNeeded.bind(this);
```

Comment:
It seems weird to me that we create this bound function after we call render, yet use this bound function inside of `_fetchElements` (albeit asynchronously). I think instead of using a "saved" bound function like this, we should just do `window.addEventListener('appopened', this);` and use `handleEvent`. That way you can just `removeEventListener(..., this)` from anywhere and don't have to create a handle for a bound function.